features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/ExploreViewModel.kt

```diff
@@ -390,6 +400,52 @@ class ExploreViewModel @Inject constructor(
         this.allMerchantLocationsJob?.cancel()
     }
 
+    fun isUserSignInDashDirect() = repository.isUserSignIn()
+
+    fun signInToDashDirect(email: String, password: String) {
+        if (repository.isUserSignIn()) {
+            _dashDirectSignIn.value = true
+            return
+        }
+
+        viewModelScope.launch {
```

Comment:
Just make `signInToDashDirect` suspend and return the result directly. See below.